pandas/core/dtypes/cast.py

```diff
@@ -329,7 +342,9 @@ def maybe_cast_result_dtype(dtype: DtypeObj, how: str) -> DtypeObj:
     return dtype
 
 
-def maybe_cast_to_extension_array(cls: Type["ExtensionArray"], obj, dtype=None):
+def maybe_cast_to_extension_array(
+    cls: Type["ExtensionArray"], obj: ArrayLike, dtype: Optional[ExtensionDtype] = None
+) -> ArrayLike:
```

Comment:
`ArrayLike` is a TypeVar. so the return type is typed to be the same as obj.

I _think_ mypy is accepting this since np.ndarray resolves to `Any`

We probably need an ArrayLikeUnion, but this is perhaps out of scope here as the discussion has been started elsewhere. xref  #36100

Reply:
Shall I remove the annotation and add a TODO? 

As an aside FWIW I like the `ArrayLike`/`ArrayLikeT` idea in https://github.com/pandas-dev/pandas/pull/36100/files#r483753222

Reply:
you can leave as is for now. we have #36092 to track the issues once np.ndarray no longer resolves to Any.